src/java.base/share/classes/sun/security/ec/ECDHKeyAgreement.java

```diff
@@ -80,20 +78,26 @@ public ECDHKeyAgreement() {
     // Generic init
     private void init(Key key) throws
         InvalidKeyException, InvalidAlgorithmParameterException {
+        privateKey = null;
+        privateKeyOps = null;
+        publicKey = null;
```

Comment:
The fields should be initialized to null, so I don't think you need these lines.

Reply:
```
KeyAgreement ka = KeyAgreement.getInstance("ECDH");
ka.init(key1);
ka.init(key2);
```
If no those lines, when the second `init` throws exception, and the keys set by the first `init` are not cleared.
Please consider the test case `testInitWithInvalidKey` in `ECDHKeyAgreementParamValidation`.

Reply:
Yes, you are right.